specs/ExecuteScript.md

```diff
@@ -0,0 +1,222 @@
+
+
+# Description
+Using ExecuteScript user can run script in the context of a calling frame and receive the script result in the callback if provided.
+If ExecuteScript is run during the navigation it should be called after ContentLoading event.
+
+# Examples
+## DOMContentLoaded event and ExecuteScript
+DOMContentLoaded is raised when the initial HTML document has been parsed.
+
+## .NET, WinRT
+```c#
+    webView.CoreWebView2.FrameCreated += (sender, args) =>
+    {
+        args.Frame.DOMContentLoaded += (frameSender, DOMContentLoadedArgs) =>
+        {
+            args.Frame.ExecuteScriptAsync(
+                "let content = document.createElement(\"h2\");" +
+                "content.style.color = 'blue';" +
+                "content.textContent = \"The url of the iframe is \" + window.location.href.toString();" +
+                "document.body.appendChild(content);");
+        };
+    };
+```
+
+## Win32 C++
+```cpp
+    CHECK_FAILURE(m_appWindow->GetWebView()->QueryInterface(IID_PPV_ARGS(&m_webView2)));
+    wil::com_ptr<ICoreWebView2_4> webview2_4 = m_webView.try_query<ICoreWebView2_4>();
+    if (webview2_4)
+    {
+        CHECK_FAILURE(webview2_4->add_FrameCreated(
+            Callback<ICoreWebView2FrameCreatedEventHandler>(
+                [](ICoreWebView2* sender, ICoreWebView2FrameCreatedEventArgs* args)
+                    -> HRESULT {
+                    wil::com_ptr<ICoreWebView2Frame> webviewFrame;
+                    CHECK_FAILURE(args->get_Frame(&webviewFrame));
+                    wil::com_ptr<ICoreWebView2ExperimentalFrame> frameExperimental =
+                        webviewFrame.try_query<ICoreWebView2ExperimentalFrame>();
+                    if (frameExperimental)
+                    {
+                        frameExperimental->add_DOMContentLoaded(
+                            Callback<
+                                ICoreWebView2ExperimentalFrameDOMContentLoadedEventHandler>(
+                                [](
+                                    ICoreWebView2Frame* frame,
+                                    ICoreWebView2DOMContentLoadedEventArgs* args) -> HRESULT {
+                                    wil::com_ptr<ICoreWebView2ExperimentalFrame> frameExperimental;
+                                    frame->QueryInterface(IID_PPV_ARGS(&frameExperimental));
+
+                                    frameExperimental->ExecuteScript(
+                                        LR"~(
+                                        let content = document.createElement("h2");
+                                        content.style.color = 'blue';
+                                        content.textContent = "The url of the iframe is " + window.location.href.toString();
+                                        document.body.appendChild(content);
+                                        )~",
+                                        Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
```

Comment:
Because it's not optional in C++. Same as main ICoreWebView2::ExecuteScript. You don't see it in C# because C# expresses the method as a Task which you can choose to `await`.

Reply:
Either replace with nullptr because we don't do anything, or update sample code to do something interesting?

Add comment to callback explaining it can be nullptr.

@david-risney update IDL to note that async callback completion handlers (and probably event?) is optional: add unique attribute